Add ISO8583-connector-example.md (#1234)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/ISO8583-connector/ISO8583-connector-example.md

# ISO8583 Connector Example

> Given below is a sample scenario that demonstrates how to WSO2 ISO8583 connector send an ISO8583 message to the financial networks using WSO2 micro integrator.

Reviewer: ```suggestion
Given below is a sample scenario that demonstrates how the WSO2 ISO8583 Connector sends an ISO8583 message to financial networks using WSO2 Micro Integrator.
```

Author: Updated.

## What you'll build

> There are many such standards exist. The ISO8583 message format is used for financial transaction, such as ATM, POS, Credit Card, Mobile Banking, Internet Banking, KIOSK, e-commerce etc. transactions.

Reviewer: ```suggestion
The ISO8583 message format is used for financial transactions such as ATM, POS, Credit Card, Mobile Banking, Internet Banking, KIOSK, e-commerce, etc. transactions.
```

Removed the first line. I think it's not required.

Author: Done.

> The financial transaction is communication between two systems through a socket connection. After connection established, each system can send messages in ISO 8583 format which commonly will be requested and the other system will response. 

Author: Done.

> For an example the purchase made in a store may travel from the merchant terminal through another require a network or networks to the issuing bank where the card holders account is held.

Reviewer: ```suggestion
For an example the purchase made in a store may travel from the merchant terminal through another. This requires a network or networks to the issuing bank where the card holder's account is held.
```

Author: Done.

> Cardholder-originated transactions include purchase, withdrawal, deposit, refund, reversal, balance inquiry, payments and inter-account transfers. ISO 8583 also defines system-to-system messages for secure key exchanges, reconciliation of totals, and other administrative purposes.The response or authorizing or decline the transaction needs to be returned by the same route to the terminal.

Reviewer: ```suggestion
Cardholder-originated transactions include purchase, withdrawal, deposit, refund, reversal, balance inquiry, payments, and inter-account transfers. ISO8583 also defines system-to-system messages for secure key exchanges, reconciliation of totals, and other administrative purposes. The response on authorizing or declining the transaction needs to be returned by the same route to the terminal.
```

Author: Updated.

> Given below is a sample API that illustrates how you can configure with the `init` operation and then use the `iso8583.sendMessage` operation for the send an ISO8583 message to the financial transactions.

Reviewer: ```suggestion
Given below is a sample API that illustrates how you can configure ISO8583 with the `init` operation and then use the `iso8583.sendMessage` operation to send an ISO8583 message for the financial transactions.
```

Author: Done.

**initialize the connector**
````xml
<iso8583.init>
       <serverHost>localhost</serverHost>
       <serverPort>5010</serverPort>
</iso8583.init>
````
**Properties**

| Property        | Description |
| ------------- |-------------|
| serverHost   | The host of the server. |
| serverPort   | The port on which the server can start listening for messages. |

**Sending an ISO8583 message**

To send an ISO8583 message, you need to use the <iso8583.sendMessage/> operation in your configuration. You can use a REST client to send an XML message, set the Content-Type header in the REST client as application/xml, and then POST the body in XML format.

The following diagram illustrates how to send an ISO8583 message using a REST client.

> <img src="/assets/img/connectors/ISO8583-connector.png" title="ISO8583 Connector" width="800" alt="ISO8583 Connector"/>

Reviewer: Shall we add some description of what exactly is happening in the diagram?

Author: Added.

## Configure the connector in WSO2 Integration Studio

Follow these steps to set up the ESB Solution Project and the Connector Exporter Project.

{!references/connectors/importing-connector-to-integration-studio.md!}

1. Our project would look similar to the following (source view).

    ```
   <?xml version="1.0" encoding="UTF-8"?>
   <api context="/sendiso" name="sendiso" xmlns="http://ws.apache.org/ns/synapse">
       <resource methods="POST">
           <inSequence>
               <log>
                   <property name="status" value="Sending_an_ISO8583_Messsage"/>
               </log>
               <iso8583.init>
                   <serverHost>localhost</serverHost>
                   <serverPort>5010</serverPort>
               </iso8583.init>
               <iso8583.sendMessage/>
               <respond/>
           </inSequence>
           <outSequence>
               <log/>
               <send/>
           </outSequence>
           <faultSequence/>
       </resource>
   </api>
        ```
2. Right-click on the Composite Application Project and click on **Export Project Artifacts and Run**. Select **Run on Micro Integrator**.

> 3. Micro Integrator will be started and the composite application will be deployed. You can further refer to the application deployed through the CLI tool. Make sure you first export the PATH as below.

Author: Done.

    ```
    $ export PATH=/path/to/mi/cli/directory/bin:$PATH\
    ```
{!/references/connectors/exporting-artifacts.md!}

## Deployment

Follow these steps to deploy the exported CApp in the Enterprise Integrator Runtime. 

{!references/connectors/deploy-capp.md!}
    
## Testing

> 1. Login in to the Micro Integrator CLI tool.

Reviewer: ```suggestion
1. Log in to the Micro Integrator CLI tool.
```

Author: Done.
